toonz/sources/toonz/startuppopup.cpp

```diff
@@ -0,0 +1,889 @@
+
+// Qt includes
+#include <QHBoxLayout>
+#include <QComboBox>
+#include <QPushButton>
+#include <QMainWindow>
+#include <QApplication>
+#include <QDesktopWidget>
+#include <QMessageBox>
+#include <QTextStream>
+#include <QFrame>
+#include <QGroupBox>
+
+using namespace std;
+using namespace DVGui;
+
+namespace {
+
+// the first value in the preset list
+const QString custom = QObject::tr("<custom>");
+
+QString removeZeros(QString srcStr) {
+  if (!srcStr.contains('.')) return srcStr;
+
+  for (int i = srcStr.length() - 1; i >= 0; i--) {
+    if (srcStr.at(i) == '0')
+      srcStr.chop(1);
+    else if (srcStr.at(i) == '.') {
+      srcStr.chop(1);
+      break;
+    } else
+      break;
+  }
+  return srcStr;
+}
+}  // namespace
+
+//=============================================================================
+/*! \class StartupPopup
+                \brief The StartupPopup class provides a modal dialog to
+   bring up recent files or create a new scene.
+
+                Inherits \b Dialog.
+*/
+//-----------------------------------------------------------------------------
+
+StartupPopup::StartupPopup()
+    : Dialog(TApp::instance()->getMainWindow(), true, true, "StartupPopup") {
+  setWindowTitle(tr("OpenToonz Startup"));
+
+  m_projectBox = new QGroupBox(tr("Choose Project"), this);
+  m_sceneBox   = new QGroupBox(tr("Create a New Scene"), this);
+  m_recentBox  = new QGroupBox(tr("Open Scene"), this);
+  m_projectBox->setSizePolicy(QSizePolicy::Expanding, QSizePolicy::Expanding);
+  m_nameFld                 = new LineEdit(this);
+  m_pathFld                 = new FileField(0);
+  m_sceneNameLabel          = new QLabel(tr("Scene Name:"));
+  m_widthLabel              = new QLabel(tr("Width:"), this);
+  m_widthFld                = new DVGui::MeasuredDoubleLineEdit(0);
+  m_heightLabel             = new QLabel(tr("Height:"), this);
+  m_heightFld               = new DVGui::MeasuredDoubleLineEdit(0);
+  m_dpiLabel                = new QLabel(tr("DPI:"), this);
+  m_dpiFld                  = new DoubleLineEdit(0, 120);
+  m_resXLabel               = new QLabel(tr("X"), this);
+  m_resXFld                 = new DVGui::DoubleLineEdit(this);
+  m_resYFld                 = new DVGui::DoubleLineEdit(this);
+  m_resTextLabel            = new QLabel(tr("Resolution:"), this);
+  m_fpsLabel                = new QLabel(tr("Frame Rate:"), this);
+  m_fpsFld                  = new DoubleLineEdit(0, 24.0);
+  m_cameraSettingsWidget    = new CameraSettingsWidget(false);
+  m_presetCombo             = new QComboBox(this);
+  m_unitsCB                 = new QComboBox(this);
+  m_addPresetBtn            = new QPushButton(tr("Add"), this);
+  m_removePresetBtn         = new QPushButton(tr("Remove"), this);
+  m_showAtStartCB           = new QCheckBox(tr("Show this at startup"), this);
+  QPushButton *createButton = new QPushButton(tr("Create Scene"), this);
+  QPushButton *newProjectButton = new QPushButton(tr("New Project..."), this);
```

Comment:
It is not actually a problem, but IMO it looks better if you can modify them in a consistent way by;
- setting `this` instead of `0` for the parent widget
- removing `DVGui::` namespace qualifiers